### PR TITLE
Fix #2971 Script completion proposal

### PR DESCRIPTION
**Pull request by williamthome**
### Description

Fix #2971

An improved version of the completion proposal for the `bin/zotonic` command.
@mworrell could you check if it works on your Mac?

### Checklist

- [ ] documentation updated
- [ ] tests added
- [x] no BC breaks

### CODE REVIEW

**Comment by mworrell** [comment]
Works on macOS!

```bash
PoToi:zotonic-master-merge marc$ bin/zotonic 
addsite          config           connectdb        dispatch         load             restartsite      shell            sitedir          start_nodaemon   stop             wait             
compile          configfiles      createdb         etop             logtail          rpc              siteconfig       sitetest         startsite        stopsite         
compilefile      configtest       debug            flush            restart          runtests         siteconfigfiles  start            status           update           
PoToi:zotonic-master-merge marc$ bin/zotonic 
```

File: apps/zotonic_launcher/bin/zotonic [hunk 1]
```diff
@@ -17,6 +17,15 @@ export ZOTONIC_BIN
...
 
 case "$1" in
+    completion)
+        COMMAND_NAMES="$(${ZOTONIC_BIN}/zotonic.escript command_names)"
+        # TODO: Check OS (this works in Ubuntu 21.10)
+        SYS_COMPLETION_DIR="/etc/bash_completion.d"
+        OUTPUT="$SYS_COMPLETION_DIR/zotonic-completion.bash"
+        # TODO: Completion for options
+        sudo sh -c "printf '#/usr/bin/env bash\n\n_zotonic_completions()\n{\n\tif [ \"\${#COMP_WORDS[@]}\" != \"2\" ]; then\n\t\treturn\n\tfi\n\n\tCOMPREPLY=(\$(compgen -W %s \"\${COMP_WORDS[1]}\"))\n}\n\ncomplete -F _zotonic_completions zotonic' '$COMMAND_NAMES' >| '$OUTPUT'"
+        sudo bash -c "source $OUTPUT"
```
Comment (mmzeeman) [inline apps/zotonic_launcher/bin/zotonic]: Is sudo this really necessary?

**Review by mmzeeman — COMMENTED** [review]
Is that sudo really necessary?

**Review by vkatsuba — COMMENTED** [review]
Minor: I suppose this commit https://github.com/zotonic/zotonic/pull/2976/commits/e14f096654ef1bfdce3bddba6533ddcd4d8fb5dc should be changed, please use `rebase` with `master` instead of `merge` with `master`.

File: apps/zotonic_launcher/bin/zotonic [hunk 2]
```diff
@@ -18,12 +18,42 @@ cd -- "${ZOTONIC}" || \exit 1
...
-        # TODO: Completion for options
-        printf "#/usr/bin/env bash\n\n_zotonic_completions()\n{\n\tif [ \"\${#COMP_WORDS[@]}\" != \"2\" ]; then\n\t\treturn\n\tfi\n\n\tCOMPREPLY=(\$(compgen -W %s \"\${COMP_WORDS[1]}\"))\n}\n\ncomplete -F _zotonic_completions zotonic" "$COMMAND_NAMES" >| "$OUTPUT"
-        bash -c "source $OUTPUT"
+        unameOut="$(uname -s)"
+        case "${unameOut}" in
+            Linux*)     machine=Linux;;
+            Darwin*)    machine=Mac;;
+            CYGWIN*)    machine=Cygwin;;
+            MINGW*)     machine=MinGw;;
+            *)          machine="UNKNOWN:${unameOut}"
```
Comment (vkatsuba) [inline apps/zotonic_launcher/bin/zotonic:27]: Maybe also need add: `SunOS`, `GNU`, `FreeBSD`, `NetBSD`, `OpenBSD`, `DragonFly`, `Msys`. All those platforms can use `shell` too.

File: apps/zotonic_launcher/bin/zotonic (hunk 1 above)
Comment (vkatsuba) [reply apps/zotonic_launcher/bin/zotonic]: I suppose it was in previous commits what was changed and `sudo` was removed.

Comment (vkatsuba) [reply apps/zotonic_launcher/bin/zotonic]: Yep, it was change at https://github.com/zotonic/zotonic/pull/2976/commits/c1f71067627b1720759b58e0f9954734f5f35323

**Comment by mmzeeman** [comment]
Thanks a lot.

**Comment by mworrell** [comment]
@williamthome If you want, you could add your name to the CONTRIBUTORS list.  With or without an email address.

File: apps/zotonic_launcher/bin/zotonic (hunk 2 above)
Comment (seannaswell) [reply apps/zotonic_launcher/bin/zotonic:27]: Thanks for catching this @[vkatsuba](https://github.com/vkatsuba). Currently using on several FreeBSD machines, and possibly DragonFly in the near future.
> ❯ uname -s
FreeBSD

> #uname -s
DragonFly

**Pull request merged**